helm/cluster-vsphere/templates/copy-secret-hook-job.yaml

```
@@ -0,0 +1,134 @@
+    "helm.sh/hook-weight": "-2"
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+subjects:
+  - kind: ServiceAccount
+    name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+    namespace: "{{ $.Release.Namespace }}"
+roleRef:
+  kind: Role
+  name: "{{ include "resource.default.name" $ }}-read-secret"
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: "{{ include "resource.default.name" $ }}-create-secret"
+  namespace: "{{ .Values.userContext.secretRef.secretNamespace }}"
+  annotations:
+    {{- include "copyHookAnnotations" $ | nindent 4 }}
+    "helm.sh/hook-weight": "-2"
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+subjects:
+  - kind: ServiceAccount
+    name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+    namespace: "{{ $.Release.Namespace }}"
+roleRef:
+  kind: Role
+  name: "{{ include "resource.default.name" $ }}-create-secret"
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+  namespace: "{{ $.Release.Namespace }}"
+  annotations:
+    {{- include "copyHookAnnotations" $ | nindent 4 }}
+    "helm.sh/hook-weight": "-1"
+  labels:
+    {{- include "labels.common" $ | nindent 4 }}
+spec:
+  ttlSecondsAfterFinished: 86400 # 24h
+  template:
+    metadata:
+      name: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+      namespace: "{{ $.Release.Namespace }}"
+      labels:
+        {{- include "labels.common" $ | nindent 8 }}
+    spec:
+      restartPolicy: Never
+      serviceAccountName: "{{ include "resource.default.name" $ }}-copy-secret-hook"
+      securityContext:
+        runAsUser: {{ include "securityContext.runAsUser" $ }}
+        runAsGroup: {{ include "securityContext.runAsGroup" $ }}
+      containers:
+        - name: copy-secret-hook
+          image: "{{ .Values.kubectlImage.registry }}/{{ .Values.kubectlImage.name }}:{{ .Values.kubectlImage.tag }}"
+          command:
+            - "/bin/sh"
+            - "-xc"
+            - |
+              # _cred=$(kubectl get secret {{ .Values.userContext.secretRef.secretName }} -n "{{ .Values.userContext.secretRef.secretNamespace }}" -ojson |\
+              #          jq -r '.data | "\(.username | @base64d)\n\(.password | @base64d)"')
+              # [ $? -ne 0 ] && echo "can't get the credentials" >&2 && exit 1
+              # _username=$(echo $_cred | head -1)
+              # _password=$(echo $_cred | tail -1)
+              kubectl get secret "{{ .Values.userContext.secretRef.secretName }}" -n "{{ .Values.userContext.secretRef.secretNamespace }}" -ojson |\
+                      jq 'del(.metadata.creationTimestamp, .metadata.uid, .metadata.resourceVersion, .metadata.namespace, .metadata.annotations)' |\
+                      jq '.metadata.name="{{ include "resource.default.name" $ }}"' |\
```

How about this?
```suggestion
                      jq '.metadata.name="{{ include "resource.default.name" $ }}-{{ .Values.userContext.secretRef.secretName }}"' |\
```

It would be good to also label this secret with `giantswarm.io/managed-by: cluster-vsphere-pre-install-hook` so we know where it comes from